src/engine/archive.cpp

```diff
@@ -138,11 +138,11 @@ namespace CityFlow {
             vehicle->controllerInfo.leader  = getNewPointer(newPool, vehicle->controllerInfo.leader);
             vehicle->controllerInfo.blocker = getNewPointer(newPool, vehicle->controllerInfo.blocker);
 
-            std::shared_ptr<LaneChange> laneChange = vehicle->laneChange;
```

Comment:
Why do you need to change the pointer to a ref?  It seems that a lot of changes in the whole code

Reply:
The reason why I used pointers is to use polymorphism to support different lane changing policies (For example, the built-in SimpleLaneChange is a subclass of LaneChange). 

But I think it is a overdesign Now. Since we have only one built-in policy, the polymorphism only increases the complexity, (especially those with checkpoint). And in the near future, we will not introduce other lane changing policy. For simplicity, I cancelled this design. 

If users need to control the lane chaning behavior, they could use python api.

Reply:
Great!